multimodal/setup.py

```diff
@@ -55,23 +55,31 @@
     "jinja2>=3.0.3,<3.2",
     "tensorboard>=2.9,<3",
     "pytesseract>=0.3.9,<0.3.11",
-    "PyMuPDF<=1.21.1",
 ]
 
 install_requires = ag.get_dependency_version_ranges(install_requires)
 
+tests_require = [
+    "black>=22.3,<23.0",
+    "isort>=5.10",
+    "datasets>=2.3.2,<=2.3.2",
+    "onnx>=1.13.0,<1.14.0",
+    "onnxruntime>=1.13.0,<1.14.0;platform_system=='Darwin'",
+    "onnxruntime-gpu>=1.13.0,<1.14.0;platform_system!='Darwin'",
+    "tensorrt>=8.5.3.1,<8.5.4;platform_system=='Linux'",
+]
+
 extras_require = {
-    "tests": [
-        "black>=22.3,<23.0",
-        "isort>=5.10",
-        "datasets>=2.3.2,<=2.3.2",
-        "onnx>=1.13.0,<1.14.0",
-        "onnxruntime>=1.13.0,<1.14.0;platform_system=='Darwin'",
-        "onnxruntime-gpu>=1.13.0,<1.14.0;platform_system!='Darwin'",
-        "tensorrt>=8.5.3.1,<8.5.4;platform_system=='Linux'",
+    "PyMuPDF": [
+        "PyMuPDF<=1.21.1",
     ]
 }
 
+for test_package in ["PyMuPDF"]:
+    tests_require += extras_require[test_package]
+
+extras_require["tests"] = tests_require
```

Comment:
What's the benefits of adding `tests_require `? It looks more complicated than using only `extras_require `.

Reply:
Adding `PyMuPDF ` into `tests_require` is kind of confusing since it's not only for test.

Reply:
it is required for tests to pass, therefore I've put it in `tests_require`. Do you have another suggestion?

Reply:
Got it. Maybe we can put `onnx` and `tensorrt` into another category in future.